lib/wrapperService.ts

```diff
@@ -0,0 +1,175 @@
+import { ContractWrapperBase } from "./contractWrapperBase";
+import ContractWrapperFactory from "./contractWrapperFactory.js";
+import { LoggingService } from "./loggingService";
+import { AbsoluteVote, AbsoluteVoteWrapper } from "./wrappers/absoluteVote.js";
+import { ContributionReward, ContributionRewardWrapper } from "./wrappers/contributionreward.js";
+import { DaoCreator, DaoCreatorWrapper } from "./wrappers/daocreator.js";
+import { GenesisProtocol, GenesisProtocolWrapper } from "./wrappers/genesisProtocol.js";
+import { GlobalConstraintRegistrar, GlobalConstraintRegistrarWrapper } from "./wrappers/globalconstraintregistrar.js";
+import { SchemeRegistrar, SchemeRegistrarWrapper } from "./wrappers/schemeregistrar.js";
+import { TokenCapGC, TokenCapGCWrapper } from "./wrappers/tokenCapGC.js";
+import { UpgradeScheme, UpgradeSchemeWrapper } from "./wrappers/upgradescheme.js";
+import { VestingScheme, VestingSchemeWrapper } from "./wrappers/vestingscheme.js";
+import { VoteInOrganizationScheme, VoteInOrganizationSchemeWrapper } from "./wrappers/voteInOrganizationScheme.js";
+
+/**
+ * An object with property names being a contract key and property value as the
+ * corresponding wrapper factory (ContractWrapperFactory<TWrapper).
+ */
+export interface ArcWrapperFactories {
```

Comment:
honestly this seems overly complex. maybe we should discuss a bit more in person the need for all these layers and modes of interaction with contracts

Reply:
The typings work nicely, and the resulting code, as used by the consumer, is very clean.